feat: provide props for ariaLabel - < Toast />, < AlertBanner />, < Combobox /> Fixes: #1642

viditagrawal56 opened the pull request:
<!--

Please go over the checklist and make sure all conditions are met.

--->


#### Description
-Fixes: #1642
-Provided AriaLabel props to override corresponding AriaLabel for i18n purposes
-No new Components were added.

#### Basic

- [X] Used plop (`npm run plop`) to create a new component.
- [X] PR has description.
- [X] New component is functional and uses Hooks.
- [X] Component is written in TypeScript.

#### Style

- [X] Styles are added to `NewComponent.modules.scss` file inside the `NewComponent` folder.
- [X] Component uses CSS Modules.
- [X] Design is compatible with [Monday Design System](https://design.monday.com/).
- [X] Styles are defined using [monday-ui-style](https://github.com/mondaycom/monday-ui-style) tokens
- [X] Displays correctly in all the themes

#### Storybook

- [X] Stories were added to `/src/components/NewComponent/__stories__/NewComponent.stories.mdx` file.
- [X] Stories include all flows of using the component.
- [X] Stories implemented using [vibe-storybook-components](https://github.com/mondaycom/vibe-storybook-components) components and tokens.

#### Tests

- [X] Tests are compliant with [TESTING_README.md](TESTING_README.md) instructions.

#### Accessibility

- [X] Component is accessible.
- [X] Component is compatible with [Vibe Accessibility Guidelines](https://style.monday.com/?path=/docs/foundations-accessibility--page)

SergeyRoyt commented on src/components/AlertBanner/AlertBanner.tsx:
```diff
@@ -122,7 +122,7 @@ const AlertBanner: VibeComponent<AlertBannerProps> & {
               kind={Button.kinds.TERTIARY}
               color={isDarkBackground ? Button.colors.ON_INVERTED_BACKGROUND : Button.colors.ON_PRIMARY_COLOR}
-              ariaLabel="close-toast"
+              closeButtonAriaLabel="Close"
```
You got it a bit wrong. Here it should be `ariaLabel={closeButtonAriaLabel}` and `closeButtonAriaLabel` should be a prop of `AlertBanner` with default value = `Close`. This way users will be able to pass other ariaLabel into the component e.g. when localization is needed and user is using the system in, let's say, German

SergeyRoyt commented on src/components/Button/Button.tsx:
Please, revert these changes in Button component

viditagrawal56 replied:
done!

SergeyRoyt commented on src/components/Combobox/Combobox.tsx:
```diff
@@ -43,7 +43,7 @@ export interface ComboboxProps extends VibeComponentProps {
    */
   withCategoriesDivider?: boolean;
-  size?: typeof BASE_SIZES[keyof typeof BASE_SIZES];
+  size?: (typeof BASE_SIZES)[keyof typeof BASE_SIZES];
```
Doesn't pass lint

viditagrawal56 replied:
Fixed!

SergeyRoyt commented on src/components/Search/Search.tsx:
```diff
@@ -27,7 +27,7 @@ export interface SearchProps extends VibeComponentProps {
   autoComplete?: string;
   /* BASE_SIZES is exposed on the component itself */
-  size?: typeof BASE_SIZES[keyof typeof BASE_SIZES];
+  size?: (typeof BASE_SIZES)[keyof typeof BASE_SIZES];
```
Doesn't pass lint

viditagrawal56 replied:
Fixed!

SergeyRoyt commented on src/components/Combobox/Combobox.tsx:
```diff
@@ -284,7 +284,7 @@ const Combobox: React.FC<ComboboxProps> & {
             wrapperClassName={cx(styles.comboboxSearchWrapper, searchWrapperClassName)}
             className={styles.comboboxSearch}
-            inputAriaLabel="Search for content"
+            searchInputAriaLabel="Search for content"
```
Same here. It should be `inputAriaLabel={searchInputAriaLabel}` and searchInputAriaLabel should be a prop of Combobox with default value = `Search for content`

viditagrawal56 replied:
Fixed!

SergeyRoyt commented on src/components/Search/Search.tsx:
Please, revert changes on this component

viditagrawal56 replied:
Done!

SergeyRoyt commented on src/components/TextField/TextField.tsx:
Should be reverted

viditagrawal56 replied:
Done!

SergeyRoyt reviewed (changes requested):
Let's fix the notes I wrote you 🙏

SergeyRoyt commented on src/components/Toast/Toast.tsx:
```diff
@@ -150,7 +150,7 @@ const Toast: FC<ToastProps> & { types?: typeof ToastType; actionTypes?: typeof T
             kind={Button.kinds.TERTIARY}
             color={Button.colors.ON_PRIMARY_COLOR}
-            ariaLabel="close-toast"
+            closeButtonAriaLabel="Close"
```
Same here

viditagrawal56 replied:
Fixed!

viditagrawal56 commented:
@SergeyRoyt  Sorry for the obvious mistakes 😬
Will fix this ASAP

viditagrawal56 commented:
@SergeyRoyt Hey! Thanks for the review and helping me fix the silly mistakes I made.
I am a beginner so I am learning a lot!
Let me know if you need any more changes 😄

SergeyRoyt reviewed (changes requested):
Well done! 😉
Now the only thing left is to update and push snapshot tests - we are using them to track the changes in our components - you just need to run `npm run test:update`, check it and commit and push
I also see that one jest test which depends on the previous value of aria-label is now failing - it's not the best practice to use aria-label to get the element in jest, so I propose to replace it with retrieving element by it's data-testid, in this case you should refer to `ComponentDefaultTestId.Button`

viditagrawal56 commented:
Hey @SergeyRoyt, I am a bit confused as to how to change toast-tests.jest.js
From what I understand from ComponentDefaultTestId.Button, the test ids are present inside the test/constants.ts inside the ComponentDefaultTestId Enum.
So I added the ComponentDefaultTestId.TOAST_BUTTON accordingly as seen in the screenshot below.

![image](https://github.com/mondaycom/monday-ui-react-core/assets/52532308/a8032941-f4fc-420f-8d55-bba3ee80df7c)

And I changed the toast-test.jest.js file accordingly as seen in the screenshot below.

![image](https://github.com/mondaycom/monday-ui-react-core/assets/52532308/e9c503c8-6041-494f-b272-f3991bb99e7d)

I ran npm run test:update and all the test cases passed.
Is my implemntation correct or do I need to change anything let me know so I can commit my changes 🙏

viditagrawal56 commented:
> Hey, @viditagrawal56 ! It's close, but not a perfect solution, cause we are currently using `ComponentDefaultTestId.TOAST_BUTTON` inside `ToastButton` component - button which you can see on the screenshot ![image](https://user-images.githubusercontent.com/104433616/274642685-fcdae7a1-a619-4e45-9337-6817919e811f.png)
> 
> So in your solution two elements would have same data-testid, which in general should be avoided
> 
> And in this close-button we currently don't pass any data-testid specifically - so it just gets Button default data-testid, that's why I told you to use `ComponentDefaultTestId.BUTTON`
> 
> But indeed the more correct solution would be to provide it a separate data-testid - it might break some existing tests for somebody using the library, but we don't care about it this much, as it only tests. So let's add a new value to `ComponentDefaultTestId` and call it `TOAST_CLOSE_BUTTON`, then let's pass it inside this close Button and use it in the tests to retrieve it

Sure! Thanks for the reply. I'll implement it ASAP.

SergeyRoyt commented:
Hey, @viditagrawal56 !
It looks amazing, thanks!! 👑
I'd only ask you to resolve the merge conflicts and fix lint errors - for lint you can just run `npm run fix-lint` 🙏

viditagrawal56 commented:
Hey @SergeyRoyt,
I am so sorry for bothering you again but I just wanted to see if I am doing something wrong.
I am a beginner and this is the first time I am solving a merge conflict.
I did git pull in vs code so that I can see the conflicts in the vs code editor itself.
Fixed the conflicts and ran npm run fix-lint.
But now vs code shows so many staged changes 😨

![image](https://github.com/mondaycom/monday-ui-react-core/assets/52532308/bfa57f01-896b-48a4-99a4-c5d8ef97fdd4)

Am I doing something wrong? Please guide me 🙏

SergeyRoyt commented:
> Hey @SergeyRoyt, I am so sorry for bothering you again but I just wanted to see if I am doing something wrong. I am a beginner and this is the first time I am solving a merge conflict. I did git pull in vs code so that I can see the conflicts in the vs code editor itself. Fixed the conflicts and ran npm run fix-lint. But now vs code shows so many staged changes 😨
> 
> ![image](https://user-images.githubusercontent.com/52532308/275346202-bfa57f01-896b-48a4-99a4-c5d8ef97fdd4.png)
> 
> Am I doing something wrong? Please guide me 🙏

Hey @viditagrawal56 !
It seems like you doing it right so far. You just need to solve conflicts in two first files:
- For Toast file open merge conflict editor and solve it line by line
- For toast snapshots just take one of the versions and then run `npm run test:update` - it will just generate relevant snapshots
Than u'll need to stage both of the files
Regarding the other changes - it should be fine, you should have just pulled the latest master and then merge. So just commit and push them with the solved conflict files
Then check in this PR changes if u haven't accidentally bring any unrelated changes

viditagrawal56 commented:
@SergeyRoyt Ok will give it a try!
Thanks for helping me 😄

viditagrawal56 commented:
@SergeyRoyt made the commit. 
Did I mess up something?

viditagrawal56 commented:
I think I made a huge mistake.
I should have just staged the two conflicting files that I fixed and then commit.

SergeyRoyt commented:
> I think I made a huge mistake. I should have just staged the two conflicting files that I fixed and then commit. Should i revert the commit?

@viditagrawal56 
No worries, man! All is fixable :) 
You can just hard reset to the latest commit before merge and try again
This time don't run lint on all of the files - that's the problem, my bad, sorry 😝Just configure `prettier` on save and prettify files you've been working on
Regarding the rest - it should be exactly how you did it
- fix lint issues in your files with prettier
- pull master & merge it to your branch
- solve Toast conflicts
- update snapshots to fix toast.snapshots conflicts
- stage everything, commit and **force** push
Then check again if you see correct files in this PR

viditagrawal56 commented:
Hey @SergeyRoyt Thanks for helping me!
I think everything is right this time.
Do I need to change anything?

viditagrawal56 commented:
All checks passed! 😀

SergeyRoyt reviewed (approved):
You are a champ, @viditagrawal56!!
Congratulations with the first contribution to Vibe! 🥳
Thanks a lot for your determination to complete the task! 👑
Keep up the good work, keep coding and improving! 💪

viditagrawal56 commented:
> You are a champ, @viditagrawal56!! Congratulations with the first contribution to Vibe! 🥳 Thanks a lot for your determination to complete the task! 👑 Keep up the good work, keep coding and improving! 💪

@SergeyRoyt A huge thanks to you for all the guidance you provided me throughout this task. 🙌
This was my biggest contribution yet and you helped me a lot, Thank you so much!
I love this community and would definitely try my best to contribute to it! 😄
